jhub_apps/hub_client/hub_client.py

```diff
@@ -101,15 +106,98 @@ def edit_server(self, username, servername, user_options=None):
         logger.info("Now creating the server with new params", server_name=servername)
         return self._create_server(username, servername, user_options)
 
-    def _create_server(self, username, servername, user_options):
+    def _create_server(self, username: str, servername: str, user_options: UserOptions = None):
         url = f"/users/{username}/servers/{servername}"
         params = user_options.model_dump()
         data = {"name": servername, **params}
         logger.info("Creating new server", server_name=servername)
         r = requests.post(API_URL + url, headers=self._headers(), json=data)
         r.raise_for_status()
+        if is_jupyterhub_5():
+            logger.info("Sharing", share_with=user_options.share_with)
+            self._share_server_with_multiple_entities(
+                username,
+                servername,
+                share_with=user_options.share_with
+            )
+        else:
+            logger.info("Not sharing server as JupyterHub < 5.x")
         return r.status_code, servername
 
+    def _share_server(
+            self,
+            username: str,
+            servername: str,
+            share_to_user: typing.Optional[str],
+            share_to_group: typing.Optional[str],
+    ):
+        url = f"/shares/{username}/{servername}"
+        if share_to_user:
+            data = {"user": share_to_user}
+        elif share_to_group:
+            data = {"group": share_to_group}
+        else:
+            raise ValueError("None of share_to_user or share_to_group provided")
+        share_with = share_to_group or share_to_user
+        logger.info(f"Sharing {username}/{servername} with {share_with}")
+        return requests.post(API_URL + url, headers=self._headers(), json=data)
```

Comment:
Do we need some escaping here? My concern is, what if a user names their server `../../another-endpoint`? Maybe it is safe because `normalize_server_name` should strip both `.` and `/`. Maybe all that needs to be done is having a test case to ensure no path traversal.

Reply:
>Maybe it is safe because normalize_server_name should strip both . and /. 

Yes, correct. I added tests for `normalize_server_name`.